analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -1186,7 +1186,16 @@ class CompileActionUniqueingType(Enum):
     # alphabetically first output object file name
     SOURCE_REGEX = 2  # Based on source file, uniqueing by regex filter
     STRICT = 3  # Gives error in case of duplicate
+    SYMLINK = 4 # Recognizing symlinks and eliminate the duplications
 
+def is_symlink(filepath):
+    """
+    Checks whether the given file path is symlink or located in the 
+    directory containing a symlink.
+    """
+    parent_dir = os.path.abspath(filepath)
```

Comment:
As @vodorok mentioned in another comment, the "directory" and "file" components always need to be handled together. `os.path.abspath()` returns an absolute path of the given file, but the current working directory will be used as prefix.